src/day7.scala

```diff
@@ -0,0 +1,74 @@
+// using scala 3.1.0
+// using lib com.lihaoyi::pprint:0.6.6
+
+package day7
+
+import scala.util.Using
+import scala.io.Source
+import scala.annotation.tailrec
+
+case class Crabmarine(
+    horizontal: Int,
+    increasingCost: Boolean,
+    fuelCost: Int = 1
+):
+  def moveUp() =
+    if increasingCost then
+      this.copy(horizontal = horizontal + 1, fuelCost = fuelCost + 1)
+    else this.copy(horizontal = horizontal + 1)
+  def moveDown() =
+    if increasingCost then
+      this.copy(horizontal = horizontal -1, fuelCost = fuelCost + 1)
+    else this.copy(horizontal = horizontal - 1)
+
+case class Crabmada(crabmarines: List[Crabmarine]):
+
+  require(crabmarines.nonEmpty)
+
+  @tailrec
+  final def align(
+      situation: List[Crabmarine] = crabmarines,
+      fuelCost: Int = 0
+  ): Int =
+    val allTheSame = situation.forall(_.horizontal == situation.head.horizontal)
+    if allTheSame then fuelCost
+    else
+      val topHorizontal = situation.maxBy(_.horizontal)
+      val minHorizontal = situation.minBy(_.horizontal)
```

Comment:
Perhaps `maxHorizontal` to go with `minHorizontal`?

Reply:
Changed